plugins/modules/sudoers.py

```diff
@@ -199,10 +199,12 @@ def matches(self):
         return content_matches and mode_matches
 
     def content(self):
-        if self.user:
+        if self.user is not None:
             owner = self.user
-        elif self.group:
+        elif self.group is not None:
             owner = '%{group}'.format(group=self.group)
+        else:
+            raise Exception('Username (user), or groupname (group) needs to be set, but not both at the same time')
```

Comment:
How about using `required_one_of` in the argument spec instead?

Reply:
I changed to that and pushed now - that was a smart way of solving it!
(Tried to paste my test playbook but my copy/paste didn't want to play nicely on this computer)
```
TASK [Only user] *******************************************************************************************************
Monday 15 January 2024  08:52:51 +0000 (0:00:01.067)       0:00:01.093 ******** 
changed: [localhost]

TASK [Only group] ******************************************************************************************************
Monday 15 January 2024  08:52:52 +0000 (0:00:00.305)       0:00:01.399 ******** 
changed: [localhost]

TASK [Both] ************************************************************************************************************
Monday 15 January 2024  08:52:52 +0000 (0:00:00.158)       0:00:01.557 ******** 
fatal: [localhost]: FAILED! => {"changed": false, "msg": "parameters are mutually exclusive: user|group"}
...ignoring

TASK [Both Empty] ******************************************************************************************************
Monday 15 January 2024  08:52:52 +0000 (0:00:00.144)       0:00:01.702 ******** 
fatal: [localhost]: FAILED! => {"changed": false, "msg": "parameters are mutually exclusive: user|group"}
...ignoring

TASK [Both Missing] ****************************************************************************************************
Monday 15 January 2024  08:52:52 +0000 (0:00:00.148)       0:00:01.850 ******** 
fatal: [localhost]: FAILED! => {"changed": false, "msg": "one of the following is required: user, group"}
...ignoring
```

Reply:
I guess other things that use `required_one_of` have cases where empty is a valid value.
Then I suppose starting to add value validation into the available quick validations becomes a rabbit hole?
`required_one_of_with_not_empty_values`, `required_one_of_with_string_values`, etc...?

The changes are currently failing tests as when removing a rule file, I didn't envisage that you'd need to specify a user or group - similary to deleting a file with the `file` module, I guess.

Perhaps user or group should only be required if state isn't absent?

Reply:
> Perhaps user or group should only be required if state isn't absent?

Sounds like a good idea. `required_if` would be useful then.